### ISSUE TITLE
TorrentDownloadStation moves seeding files instead of copying them

### ISSUE THREAD

**Issue by samrum** [Sonarr/Sonarr#2520]
## Bug Report
When a torrent download sent to the TorrentDownloadStation client finishes downloading and is seeding, it's being moved instead of copied when it's imported. I believe the correct functionality is that when a torrent is being seeded it's supposed to be copied instead of moved.

### System Information/Logs

**Sonarr Version:**
2.0.0.5163

**Link to Log Files (debug or trace):**
https://pastebin.com/czQ1NdDt

### Additional Information
I looked into the code a bit, and it seems like when download items from TorrentDownloadStation are imported, item.CanMoveFiles isn't being set correctly. It uses the return value of an IsCompleted method which returns true when torrent status is set to seeding which is incorrect. 
https://github.com/Sonarr/Sonarr/blob/15bd181f165e5cc0176a5d5b19059f7be74f79ef/src/NzbDrone.Core/Download/Clients/DownloadStation/TorrentDownloadStation.cs#L205

Also, TorrentDownloadStation doesn't seem to be consistent with the way other torrent download clients set the value of item.CanMoveFiles. Other clients seem to set item.CanMoveFiles to the value of item.CanBeRemoved.
https://github.com/Sonarr/Sonarr/blob/19a4d3536bde8103c4e0f7414cc1819b2083ffb2/src/NzbDrone.Core/Download/Clients/uTorrent/UTorrent.cs#L137
https://github.com/Sonarr/Sonarr/blob/17cfaf170e56c80a4e0f1ec2a691b2e3513362c7/src/NzbDrone.Core/Download/Clients/Deluge/Deluge.cs#L149
https://github.com/Sonarr/Sonarr/blob/ef8b882258786ed99cc04f81e9414d039bfb0357/src/NzbDrone.Core/Download/Clients/QBittorrent/QBittorrent.cs#L116
https://github.com/Sonarr/Sonarr/blob/6a6d415625b54a7f2375d96d82a4794383d1eeb5/src/NzbDrone.Core/Download/Clients/Transmission/TransmissionBase.cs#L99

**Comment by markus101** [comment]
TLDR, this is not a bug, it is expected.

> TorrentDownloadStation doesn't seem to be consistent with the way other torrent download clients set the value of item.CanMoveFiles. Other clients seem to set item.CanMoveFiles to the value of item.CanBeRemoved.

Correct, for almost every client the file Sonarr gets from the API is actively being seeded, but, because Down Station exposes a hardlink of the file as the completed download. Later if you remove the download from Download Station the original file is deleted, but as far as I know, not the hard linked copy, which would result in that file never automatically being cleaned up.

If the file is on the same volume, the move maintains the hard link so no extra space is used. If it's going to a different volume the file will be copied and extra space is used, but that's the case regardless.

**Comment by Taloth** [comment]
Fun fact, the IsReadOnly property was split into CanMoveFiles and CanBeRemoved exactly because the logic has to be different for DownloadStation :smile:
